src/video/x11/SDL_x11keyboard.c

```diff
@@ -203,7 +203,7 @@ int X11_InitKeyboard(SDL_VideoDevice *_this)
         (void)setlocale(LC_ALL, "");
         X11_XSetLocaleModifiers(new_xmods);
 
-        data->im = X11_XOpenIM(data->display, NULL, data->classname, data->classname);
+        data->im = X11_XOpenIM(data->display, NULL, NULL, NULL);
```

Comment:
Why are we removing the classname here?

Reply:
The classname was moved out of the global video core struct and made per-window, as this allows it to be set after the video system is initialized, but before window creation, and applications may want to change the hint on-the-fly to create windows with different classes/IDs, which is perfectly valid behavior.

Using the class name for the `res_name` parameter here was generally incorrect behavior, as the executable name should explicitly be used wherever X calls for `res_name`, and using the class name here for `res_class` is going to be outdated if the application changes the class hint after video initialization.  There seems to be no harm in just leaving these as NULL, as it doesn't affect application behavior, and I can't find anywhere in the X code where these values are actually used beyond being passed back to user application callbacks, which SDL doesn't use. Every other toolkit I looked into (GTK, Qt, GLFW, SFML and the JDK) leaves these parameters NULL as well.